.github/scripts/checkKnipRegression.ts

```diff
@@ -0,0 +1,54 @@
+import * as $ from 'shelljs'
+
+const SECTION_REGEX = /[a-zA-Z ]+\([0-9]+\)/
+const CATEGORY_REGEX = /[a-zA-Z ]+/
+const PARENS_REGEX = /\(|\)| /
+
+function parseKnipOutput(knipOutput: string): Record<string, number> {
+  const lines = knipOutput.split('\n')
+  const result: Record<string, number> = {}
+  for (const line of lines) {
+    if (SECTION_REGEX.test(line)) {
+      const tokens = line.split(PARENS_REGEX)
+      const category = (line.match(CATEGORY_REGEX) as RegExpMatchArray)[0]
+      result[category] = parseInt(tokens[tokens.length - 2])
+    }
+  }
+  return result
+}
+
+function compareKnipResults(
+  mainKnipResults: Record<string, number>,
+  branchKnipResults: Record<string, number>
+): boolean {
+  let shouldFail = false
+  for (const [category, mainProblems] of Object.entries(mainKnipResults)) {
+    const branchProblems = branchKnipResults[category] ?? 0
+    console.log(`${category} Main: ${mainProblems} Branch: ${branchProblems}`)
```

Comment:
nit: what do you think of using [`console.table`](https://developer.mozilla.org/en-US/docs/Web/API/console/table) to have a slightly more readable summary?

Reply:
oh this is pretty handy! will do.